client/src/js/settings/components/Jobs/TaskField.js

```diff
@@ -44,7 +46,7 @@ export default class TaskField extends React.PureComponent {
     handleSubmit = (e) => {
         e.preventDefault();
         this.setState({pending: true}, () => {
-            this.props.onChange(this.state.value);
+            this.props.onChange(this.state.name, this.state.value);
```

Comment:
Use `this.props.name` after fixing the first change (`name` should stay in `prop`).